gpMgmt/bin/go-tools/test/integration/configure/configure_failures_test.go

```diff
@@ -0,0 +1,293 @@
+	"testing"
+)
+
+type ConfigureFailTC struct {
+	name             string
+	cliParams        []string
+	expectedOut      []string
+	expectedExitcode int
+	IsSigleHost      bool
+	IsMultiHost      bool
+}
+
+// negative test cases
+var ConfigureFailTestCases = []ConfigureFailTC{
+	{
+		name:      "configure service with empty value for --host option",
+		cliParams: []string{"--host", ""},
+		expectedOut: []string{
+			"please provide a valid input host name",
+		},
+		expectedExitcode: testutils.ExitCode1,
+		IsMultiHost:      true,
+		IsSigleHost:      true,
+	},
+	{
+		name:      "configure service with no value for --host option",
+		cliParams: []string{"--host"},
+		expectedOut: []string{
+			"flag needs an argument: --host",
+		},
+		expectedExitcode: testutils.ExitCode1,
+		IsMultiHost:      true,
+		IsSigleHost:      true,
+	},
+	{
+		name:      "configure service with empty file for --hostfile option",
+		cliParams: []string{"--hostfile", testutils.Hostfile},
+		expectedOut: []string{
+			"expected at least one host or hostlist specified",
+		},
+		expectedExitcode: testutils.ExitCode1,
+		IsMultiHost:      true,
+		IsSigleHost:      true,
+	},
+	{
+		name:      "configure service with no value for --hostfile option",
+		cliParams: []string{"--hostfile"},
+		expectedOut: []string{
+			"flag needs an argument: --hostfile",
+		},
+		expectedExitcode: testutils.ExitCode1,
+		IsMultiHost:      true,
+		IsSigleHost:      true,
+	},
+	{
+		name:      "configure service with non-existing host for --host option",
+		cliParams: []string{"--host", "host"},
+		expectedOut: []string{
+			"could not copy gp.conf file to segment hosts",
+		},
+		expectedExitcode: testutils.ExitCode1,
+		IsMultiHost:      true,
+		IsSigleHost:      true,
+	},
+	{
+		name:      "configure service with one valid host and invalid host for --host option",
+		cliParams: []string{"--host", testutils.DefaultHost, "--host", "invalid"},
+		expectedOut: []string{
+			"could not copy gp.conf file to segment hosts",
+		},
+		expectedExitcode: testutils.ExitCode1,
+		IsMultiHost:      true,
+		IsSigleHost:      true,
+	},
+	{
+		name: "configure service without any option",
+		expectedOut: []string{
+			"at least one hostname must be provided using either --host or --hostfile",
+		},
+		expectedExitcode: testutils.ExitCode1,
+		IsMultiHost:      true,
+		IsSigleHost:      true,
+	},
+	{
+		name:      "configure service with invalid option",
+		cliParams: []string{"--invalid"},
+		expectedOut: []string{
+			"unknown flag: --invalid",
+		},
+		expectedExitcode: testutils.ExitCode1,
+		IsMultiHost:      true,
+		IsSigleHost:      true,
+	},
+	{
+		name: "configure service with both host and hostfile options",
+		cliParams: []string{"--host", testutils.DefaultHost,
+			"--hostfile", "abc"},
+		expectedOut: []string{
+			"[ERROR] if any flags in the group [host hostfile] are set none of the others can be; [host hostfile] were all set",
```

Comment:
Can this be rephrased? This feels confusing reading it the first time.
Or is it missing a comma?

```suggestion
			"[ERROR] Invalid flag combination; [host hostfile] are both set",
```

Reply:
@piyushc01 I think we need to change it in source code as well.

Reply:
Yes, will change the message for an invalid combination